File: src/view/console/Screen.java
package view.console;

import model.structs.Rect;

import java.io.IOException;

public interface Screen {
    Window createWindow(Rect rect);

    Rect getRect();

    boolean paint();

    boolean forcePaint();

    static void clear(){
        try {
            new ProcessBuilder("cmd", "/c", "cls").inheritIO().start().waitFor();
        } catch (InterruptedException | IOException _) {/*I don't care!*/}
    }

    int clearScreenBuffer();
}

File: src/view/console/Screen.kt
package view.console

import model.structs.Rect
import java.io.IOException

interface Screen {
    fun createWindow(rect: Rect): Window?

    val rect: Rect

    fun paint(): Boolean

    fun forcePaint(): Boolean

    fun clearScreenBuffer(): Int

    companion object {
        @JvmStatic
        fun clear() {
            try {
                ProcessBuilder("cmd", "/c", "cls").inheritIO().start().waitFor()
            } catch (e: InterruptedException) { /*I don't care!*/
            } catch (e: IOException) {
            }
        }
    }
}
